src/circuits/runcircuit.jl

```diff
@@ -384,22 +385,31 @@ runcircuit(gates::Vector{<:Tuple}; kwargs...) =
 """
 Run the layered circuit with the possibility of having a circuit observer 
 """
-function runcircuit(M::Union{MPS,MPO}, gates::Vector{Vector{<:Tuple}}; 
+function runcircuit(M0::Union{MPS,MPO}, gates::Vector{Vector{<:Tuple}}; 
                     observer! = nothing,
+                    move_sites_back_before_measurements::Bool = false,
                     kwargs...)
-
-  for l in 1:length(gates)-1
+  if !isnothing(observer!) && !isempty(observer!.functions) && move_sites_back_before_measurements == false
+    println("--------------")
+    println(" WARNING")
+    println("\nA custom function is being measured during the gate evolution,\nbut the MPS sites at depth D are now being restored to their \nlocation at depth D-1. If any custom measurement is dependent \non the specific site ordering of the MPS, it should take the \nre-ordering into account.")
```

Comment:
"now being restored" -> "not being restored". Also, we can suggest setting `move_sites_back_before_measurements = true`.